src/test/blockmanager_tests.cpp

```diff
@@ -139,12 +139,18 @@ BOOST_FIXTURE_TEST_CASE(blockmanager_block_data_availability, TestChain100Setup)
 
 BOOST_FIXTURE_TEST_CASE(blockmanager_readblock_hash_mismatch, TestingSetup)
 {
-    CBlockIndex* fake_index{WITH_LOCK(m_node.chainman->GetMutex(), return m_node.chainman->ActiveChain().Tip())};
-    fake_index->phashBlock = &uint256::ONE; // invalid block hash
+    CBlockIndex index;
+    {
+        LOCK(cs_main);
+        const auto tip{m_node.chainman->ActiveTip()};
+        index.nStatus = tip->nStatus;
+        index.nDataPos = tip->nDataPos;
+        index.phashBlock = tip->phashBlock + 1; // invalid block hash
```

Comment:
I think you meant to increment the hash with one, rather than doing pointer arithmetic? Currently, this looks like UB to me. I think sticking to `&uint256::ONE` here is the preferable approach, no need to allocate a new hash imo.

```suggestion
        index.phashBlock = &uint256::ONE; // mismatched block hash
```

Reply:
Good point, thanks, added you as coauthor